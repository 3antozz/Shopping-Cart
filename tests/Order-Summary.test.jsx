import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import Summary from "../src/Components/Order-Summary/Order-Summary.jsx";
import { MemoryRouter, Outlet, Routes, Route } from "react-router-dom";

describe("Summary Component", () => {
    const context = {
        products: [
            {
                id: 1,
                title: "test",
                price: 69,
                category: "...",
                description: "...",
                image: "...",
                quantity: 1,
            },
        ],
        error: false,
        loading: false,
        cartIDs: [
            {
                id: 1,
                quantity: 2,
            },
        ],
        handleAddToCart: vi.fn(),
        removeProduct: vi.fn(),
        clearCart: vi.fn(),
    };
    beforeEach(() => {
        render(
            <MemoryRouter initialEntries={["/summary"]}>
                <Routes>
                    <Route
                        path="/"
                        element={
                            <div>
                                <h1>APP ROUTE</h1>
                                <Outlet context={context} />
                            </div>
                        }
                    >
                        <Route path="summary" element={<Summary />}></Route>
                    </Route>
                </Routes>
            </MemoryRouter>
        );
    });
    it("renders summary container", () => {
        expect(screen.getByTestId(/summary-container/i)).toBeInTheDocument();
    });
    it("renders checkout card container", () => {
        expect(screen.getByTestId(/checkout-container/i)).toBeInTheDocument();
    });
    it("renders product card container", () => {
        expect(screen.getByTestId(/product-container/i)).toBeInTheDocument();
    });
    it("matches snapshot", () => {
        const context = {
            products: [
                {
                    id: 1,
                    title: "test",
                    price: 69,
                    category: "...",
                    description: "...",
                    image: "...",
                    quantity: 1,
                },
            ],
            error: false,
            loading: false,
            cartIDs: [
                {
                    id: 1,
                    quantity: 2,
                },
            ],
            handleAddToCart: vi.fn(),
            removeProduct: vi.fn(),
            clearCart: vi.fn(),
        };
        const { container } = render(
            <MemoryRouter initialEntries={["/summary"]}>
                <Routes>
                    <Route
                        path="/"
                        element={
                            <div>
                                <h1>APP ROUTE</h1>
                                <Outlet context={context} />
                            </div>
                        }
                    >
                        <Route path="summary" element={<Summary />}></Route>
                    </Route>
                </Routes>
            </MemoryRouter>
        );
        expect(container).toMatchSnapshot();
    });
});
