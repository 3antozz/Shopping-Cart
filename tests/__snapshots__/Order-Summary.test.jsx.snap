// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Summary Component > matches snapshot 1`] = `
<div>
  <div>
    <h1>
      APP ROUTE
    </h1>
    <div
      class="_summary_d70768"
      data-testid="summary-container"
    >
      <div
        class="_products_d70768"
      >
        <div
          class="_productCard_824104"
          data-testid="product-container"
        >
          <a
            data-discover="true"
            href="/shop/products/1"
          >
            <img
              alt="test"
              src="..."
            />
          </a>
          <div
            class="_input_824104"
          >
            <a
              data-discover="true"
              href="/shop/products/1"
            >
              test
            </a>
            <div>
              <h3>
                Total: 
                138
                 $
              </h3>
              <label
                for="quantity"
              />
              <button
                type="button"
              >
                -
              </button>
              <input
                id="quantity"
                max="100"
                min="1"
                type="number"
                value="2"
              />
              <button
                class="increment"
                type="button"
              >
                +
              </button>
            </div>
          </div>
          <button
            type="button"
          >
            X
          </button>
        </div>
      </div>
      <div
        class="_card_65c2c8"
        data-testid="checkout-container"
      >
        <div
          class="_info_65c2c8"
        >
          <h3>
            Subtotal:
          </h3>
          <h3>
            138
             $
          </h3>
        </div>
        <div
          class="_info_65c2c8"
        >
          <h3>
            Shipping:
          </h3>
          <h3>
            Free
          </h3>
        </div>
        <div
          class="_info_65c2c8"
        >
          <h2>
            Total:
          </h2>
          <h2>
            138
             $
          </h2>
        </div>
        <button>
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;
